enhancements/sdk-fbc-run-bundle.md

```diff
@@ -0,0 +1,291 @@
+
+3. The Operator SDK, using the data gathered from the above steps as parameters, will invoke Operator Registry APIs that will generate three major components in the form of Operator Registry-defined objects - Package, Channel & Bundle.
+
+4. The Operator SDK will convert these objects into more commonly-consumable formats like JSON/YAML blobs and combine these blobs together to generate a basic File-Based Catalog representation of the bundle image. In other words, the simplest File-Based Catalog consists of one package blob, one channel blob, and one bundle blob. These blobs represent the bundle data. This step represents the "generating FBC on the fly" scenario that we referred to in this document.
+
+5. If an index image was passed in as a parameter through the `--index-image` flag of the run bundle command, a series of additional steps take place:
+
+    - The index image will be rendered into a declarative config object, similar to the bundle image.
+    - Operator SDK will check if the bundle image is already present in the index image.
+        - If the bundle image is already present in the index image, Operator SDK will not add the bundle to the index but rather error out.
+        - If the bundle is not present in the index image, Operator SDK will add the contents of the bundle image to the existing index image and serve the resulting index image (with the bundle added to it).
+
+    - Note: If the index image that was passed is in SQLite format, Operator SDK will use a different container creation command that uses `opm registry add` to add the bundle to the index and `opm registry serve` to serve the index which is a database file on created on the pod.
+
+6. The Operator SDK will invoke more Operator Registry APIs to ensure that the FBC that was generated from the above steps is indeed a valid FBC. This steps essentially validates the generated FBC by converting the declarative config object to a model object which internally performs validations equivalent to the `opm validate <FBC>` command.
+
+7. Three unique data points will be stored in the `IndexImageCatalogCreator` object that will eventually be used in the ephemeral pod:
+
+    - FBC directory name where the FBC file resides in → `FBCDir`
+    - FBC file name that has all the contents to be served through the pod → `FBCFile`
+    - Contents of FBC file (string YAML/JSON) → `FBCContent`
+
+8. During the creation of the ephemeral pod (which is used to serve the File-Based Catalog to gRPC), the Operator SDK will invoke a set of CLI commands to create a directory and file on the pod during runtime.
+
+9. The contents of the generated File-Based Catalog will be written to the file that is created on the ephemeral pod using the following command template -
+
+    ```go
+    `mkdir -p {{ .FBCDir }} && \
+    echo '{{ .FBCContent }}' >> {{ .FBCFile }} && \
```

Comment:
We should call out here that there is a limit to the size of the `FBCContent` string. For `run bundle` cases that don't specify a catalog image to use, we're probably okay in most scenarios. However, I'm guessing that `run bundle` would fail if the upstream FBC operatorhub catalog was specified as the catalog image.

We likely need to investigate a solution to this problem. My guess is that we'd actually run the FBC catalog image and we'd inject _just_ the extra FBC that we needed to generate for the new bundle.

Reply:
@joelanford Thanks for mentioning this. Captured this as a note and a risk in the EP. 

I just ran the command with the index image set to `quay.io/operatorhubio/catalog:latest` and it failed with the error `FATA[0031] Failed to run bundle: create catalog: error creating registry pod: error creating pod: Request entity too large: limit is 3145728`. 

How can we inject just the extra FBC that we need to generate for the bundle? Earlier, you mentioned the possibility of mounting it through one or more `ConfigMaps`. Is that a way to go about it?